### DISCUSSION TITLE
How to Zoom fitz.Pixmap to precise size

### DISCUSSION THREAD

**Discussion by dmytro-balabukh** [pymupdf/PyMuPDF#1048] [Looking for help] [answered]
I need my PDF page to gently fit the borders of the Reading Window. I intend to do it like, QScrollArea doesn't activate, but it's near to it, also all proportions must be saved. In Matrix, I can't give exact height, width, so what can you suggest? 

I'm using:
PyQT5
The Window you see is:
QScrollArea, which has QLabel inside

![image](https://user-images.githubusercontent.com/52417824/118052501-8948ae80-b38b-11eb-9b7d-83bce453d128.png)

**Comment by JorjMcKie** [comment, marked as answer]
Without going into Qt details (because I am not very acquainted with it, as I mostly using wxPython):
* you want to make sure that the available area is best used by the page's pixmap: at least one of its width or height is exhausted
* call the area's dimensions `WIDTH`, resp. `HEIGHT`, and the page's dimension `width`, resp `height` (i.e. `page.rect.width`, `page.rect.height`)
* then compute the pixmap and the QImage like this

```python
if width / height < WIDTH / HEIGHT:
    zoom = HEIGHT / height
else:
    zoom = WIDTH / width
mat = fitz.Matrix(zoom, zoom)
pix = page.getPixmap(matrix=mat, alpha=False)
qimg = QImage(pix.samples, pix.width, pix.height, pix.stride, QImage.Format_RGB888)
```
